src/common/fair_mutex.h

```diff
@@ -22,6 +26,7 @@ class fair_mutex {
     cond.wait(lock, [&] {
       return my_id == unblock_id;
     });
+    locked_by = std::this_thread::get_id();
```

Comment:
i think we don't need to keep track of `locked_by` in non-debug mode. i created a change at https://github.com/ceph/ceph/pull/42622/commits/6ec153cfbf0cc08af85829df959dd86e139ac5d3 .

Reply:
Yeah, right, will fix it.